forms-flow-web/src/components/Form/EditForm/FormEdit.js

```diff
@@ -172,80 +172,109 @@ const EditComponent = () => {
+      await handleImportResponse(res, fileContent, data.action);
+    } catch (err) {
+      handleImportError(err);
+    }
+  };
+  
+  // Helper function to validate the action type
+  const isValidUploadActionType = (actionType) => {
+    if (!["validate", "import"].includes(actionType)) {
       console.error("Invalid UploadActionType provided");
       setImportLoader(false);
-      return;
+      return false;
     }
-
-    let data = {};
-    data.importType = "edit";
-    data.action = UploadActionType;
-    // Set form submission state for "import" action
-
-    if (UploadActionType === "import") {
+    return true;
+  };
+  
+  // Helper function to prepare data for the API request
+  const prepareImportData = (actionType, selectedLayoutVersion, selectedFlowVersion) => {
+    const data = {
+      importType: "edit",
+      action: actionType,
+      mapperId: processListData.id,
+    };
+  
+    if (actionType === "import") {
       setImportLoader(true);
       setFormSubmitted(true);
-      // Handle selectedLayoutVersion logic
+  
       if (selectedLayoutVersion || selectedFlowVersion) {
-        data.form = {
-          skip: selectedLayoutVersion === true,
-        };
-
-        data.workflow = {
-          skip: selectedFlowVersion === true,
-        };
+        data.form = prepareVersionData(selectedLayoutVersion);
+        data.workflow = prepareVersionData(selectedFlowVersion);
       }
     }
-
-    // Add mapperId if available
-    data.mapperId = processListData.id;
-
-    // Convert data to a JSON string for the API request
-    const dataString = JSON.stringify(data);
-    try {
-      const res = await formImport(fileContent, dataString);
-      setImportLoader(false);
-      setFormSubmitted(false);
-      const { data: responseData } = res;
-      if (responseData) {
-        const { workflow, form } = responseData;
-        setFileItems({
-          workflow: {
-            majorVersion: workflow?.majorVersion || null,
-            minorVersion: workflow?.minorVersion || null,
-          },
-          form: {
-            majorVersion: form?.majorVersion || null,
-            minorVersion: form?.minorVersion || null,
-          },
-        });
-      }
-      if (data.action === "validate") {
-        FileService.extractFileDetails(fileContent)
-          .then((formExtracted) => {
-            if (formExtracted) {
-              setFormTitle(formExtracted.formTitle);  // Set the form title if form is found
-            } else {
-              console.log("No valid form found.");
-            }
-          })
-          .catch((error) => {
-            console.error("Error extracting file details:", error); // Catch any errors in the process
-          });
+  
+    return data;
+  };
+  
+  // Helper function to prepare version data
+  const prepareVersionData = (version) => ({
+    skip: version === true,
+    ...(version !== true && version !== false && { selectedVersion: version }),
```

Comment:
here are we expecting a string value ? or boolean value ? 
if its boolean version !==true && version !==false block wont be execute @Josephalexantony-aot

Reply:
yes, correct. Updated code with checking string.